src/os/bluestore/BlueStore.cc

```diff
@@ -6525,14 +6525,27 @@ void BlueStore::_txc_finish_kv(TransContext *txc)
     finishers[n]->queue(txc->onreadable);
     txc->onreadable = NULL;
   }
-  if (txc->oncommit) {
-    finishers[n]->queue(txc->oncommit);
-    txc->oncommit = NULL;
-  }
-  while (!txc->oncommits.empty()) {
-    auto f = txc->oncommits.front();
-    finishers[n]->queue(f);
-    txc->oncommits.pop_front();
+  if (g_conf->bluestore_no_finisher && !txc->is_pipelined_io &&
```

Comment:
I think we want to use the new flag I mentioned (txc->kv_submitted_sync or similar?).. is_pipelined_io may be false for things that weren't done synchronously